R/FilterStateChoices.R

```diff
@@ -223,7 +223,11 @@ ChoicesFilterState <- R6::R6Class( # nolint
           )
         } else {
           # This handles numerics, characters, and factors.
-          call(fun_compare, varname, make_c_call(choices))
+          if (!all(na.omit(private$x) %in% choices)) {
```

Comment:
Perhaps this check should go in `FilterState$is_any_filtered`?

Reply:
no because it would return `FALSE` for `$is_any_filtered()`, hence `$get_call()` method would stop sooner 
```{R}
      if (isFALSE(private$is_any_filtered())) {
        return(NULL)
      }
````

and it would not append `$add_keep_na_call()` method at the end

Reply:
moved in here https://github.com/insightsengineering/teal.slice/pull/402/commits/2e4990ada5fab287ee7935fa3ac96848eefd13ac